### PR TITLE
Introduce Data Drainer (avocado.utils.datadrainer)

### PR DESCRIPTION
**Pull request by clebergnu**
This module is a loose generalization of the FDDrainer present in avocado.utils.process, but intended for a wider audience.

### CODE REVIEW

File: avocado/utils/datadrainer.py [hunk 1]
```diff
@@ -163,3 +163,28 @@ def data(self):
...
+            self._buffer.write(data)
+            return
+        data = self._buffer.getvalue() + data
+        lines = data.split(b'\n')
+        if not lines[-1].endswith(b'\n'):
+            self._buffer.close()
+            self._buffer = io.BytesIO()
+            self._buffer.write(lines[-1])
+        for line in lines:
+            line = line.decode(errors='replace').strip()
```
Comment (apahim) [inline avocado/utils/datadrainer.py]: Should we really `strip()` or just `rtrip('\n')`?

File: selftests/unit/test_utils_datadrainer.py [hunk 2]
```diff
@@ -74,3 +74,31 @@ def test(self):
...
+        socket_drainer.start()
+        self.socket1.send(b'1')
+        self.socket1.send(b'2')
+        self.socket1.send(b'3')
+        socket_drainer.wait()
+        self.assertEqual(socket_drainer.data, b'123')
+
+    def tearDown(self):
+        self.socket1.close()
+        self.socket2.close()
```
Comment (apahim) [inline selftests/unit/test_utils_datadrainer.py]: ok

File: avocado/utils/datadrainer.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+to be drained/processed.  A thread is started and maintained on behalf
+of the user.
+"""
+
+import threading
+
+
+class BaseDrainer:
+    """
+    Base drainer, doesn't provide complete functionality to be useful.
```
Comment (apahim) [inline avocado/utils/datadrainer.py]: Yet, you allow it to be instantiated... shouldn't this be an abstract class?

File: avocado/utils/datadrainer.py (hunk 1 above)
Comment (clebergnu) [reply avocado/utils/datadrainer.py]: You're right! Thanks for catching this.

File: avocado/utils/datadrainer.py (hunk 3 above)
Comment (clebergnu) [reply avocado/utils/datadrainer.py]: It should... something was bothering me to make this an ABC, but I kind of ignored it.  Changing it in a v2.

**Comment by clebergnu** [comment]
Closing in favor of #3142

**Pull request closed without merging**